reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServer.java

```diff
@@ -315,14 +315,23 @@ public final HttpServer httpRequestDecoder(Function<HttpRequestDecoderSpec, Http
 	 * @since 0.9.7
 	 */
 	public final HttpServer metrics(boolean enable, Function<String, String> uriTagValue) {
+		return metrics(enable, () -> configuration().defaultMetricsRecorder(), uriTagValue);
+	}
+
+	@Override
+	public final HttpServer metrics(boolean enable, Supplier<? extends ChannelMetricsRecorder> recorder) {
+		return metrics(enable, recorder, configuration().uriTagValue());
+	}
+
+	public final HttpServer metrics(boolean enable, Supplier<? extends ChannelMetricsRecorder> recorder, Function<String, String> uriTagValue) {
 		if (enable) {
 			if (!Metrics.isInstrumentationAvailable()) {
 				throw new UnsupportedOperationException(
 						"To enable metrics, you must add the dependency `io.micrometer:micrometer-core`" +
 								" to the class path first");
 			}
```

Comment:
Do we need to check for Micrometer as this is a custom recorder that might not need it?
We need it only in case of the default implementation.

Reply:
No, you are right and I now understand why the code looked like it did before my change. I'll fix.